dkeyserver/Enclave/enclave.cpp

```diff
@@ -387,7 +335,329 @@ int handle_communication_until_done(
     return ret;
 }
 
-int sgx_set_up_tls_server(char *server_port)
+static unsigned long inet_addr(const char *str)
+{
+    unsigned long lHost = 0;
+    char *pLong = (char *)&lHost;
+    char *p = (char *)str;
+    while (p)
+    {
+        *pLong++ = atoi(p);
+        p = strchr(p, '.');
+        if (p)
+            ++p;
+    }
+    return lHost;
+}
+
+int connect(int sockfd, const struct sockaddr *servaddr, socklen_t addrlen)
+{
+    int ret = -1;
+
+    if (ocall_connect(&ret, sockfd, servaddr, addrlen) == SGX_SUCCESS)
+        return ret;
+
+    return -1;
+}
+
+int create_socket(const char *server_name, uint16_t server_port)
+{
+    int sockfd = -1;
+    struct sockaddr_in dest_sock;
+    int ret = -1;
+
+    sockfd = socket(AF_INET, SOCK_STREAM, 0);
+    if (sockfd == -1)
+    {
+        log_d(TLS_SERVER "Error: Cannot create socket %d.\n", errno);
+        goto out;
+    }
+
+    dest_sock.sin_family = AF_INET;
+    dest_sock.sin_port = htons(server_port);
+    dest_sock.sin_addr.s_addr = inet_addr(server_name);
+    bzero(&(dest_sock.sin_zero), sizeof(dest_sock.sin_zero));
+
+    if (connect(
+            sockfd, (sockaddr *)&dest_sock,
+            sizeof(struct sockaddr)) == -1)
+    {
+        log_d(
+            TLS_SERVER "failed to connect to target server %d:%d (errno=%d)\n",
+            server_name,
+            server_port,
+            errno);
+        ocall_close(&ret, sockfd);
+        if (ret != 0)
+            log_d(TLS_SERVER "OCALL: error closing socket\n");
+        sockfd = -1;
+        goto out;
+    }
+    log_d(TLS_SERVER "connected to target server %s:%d\n", server_name, server_port);
+
+out:
+    return sockfd;
+}
+
+sgx_status_t get_domainkey_from_target(uint8_t *domain_key, const char *target_server_name, uint16_t target_server_port)
+{
+    log_i(TLS_SERVER "Connecting target server...\n");
+    sgx_status_t ret = SGX_ERROR_UNEXPECTED;
+
+    if (target_server_name[0] == '\0' && target_server_port == 0){
```

Comment:
also add a check for the domainkey

Reply:
how about this case:
./dkeyserver -p 111 // but w/o -i

So that it can be  removed:
&& target_server_port == 0

Reply:
done.